homeassistant/components/proxmoxve/__init__.py

```diff
@@ -143,43 +103,93 @@ def build_client() -> ProxmoxAPI:
         host_name = host_config["host"]
         coordinators[host_name] = {}
 
-        proxmox_client = hass.data[PROXMOX_CLIENTS][host_name]
+    # import to config flow
+    if DOMAIN in config:
+        for conf in config[DOMAIN]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+                )
+            )
 
-        # Skip invalid hosts
-        if proxmox_client is None:
-            continue
+    return True
 
-        proxmox = proxmox_client.get_api_client()
 
-        for node_config in host_config["nodes"]:
-            node_name = node_config["node"]
-            node_coordinators = coordinators[host_name][node_name] = {}
+async def async_setup_entry(hass: HomeAssistant, config_entry):
+    """Set up the platform."""
 
-            for vm_id in node_config["vms"]:
-                coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, vm_id, TYPE_VM
-                )
+    entry_data = config_entry.data
 
-                # Fetch initial data
-                await coordinator.async_refresh()
+    hass.data[PROXMOX_CLIENTS] = {}
 
-                node_coordinators[vm_id] = coordinator
+    host = entry_data[CONF_HOST]
+    port = entry_data[CONF_PORT]
+    user = entry_data[CONF_USERNAME]
+    realm = entry_data[CONF_REALM]
+    password = entry_data[CONF_PASSWORD]
+    verify_ssl = entry_data[CONF_VERIFY_SSL]
```

Comment:
Why not directly use these when creating a ProxmoxClient instance, instead of creating all these variables?